include/OpenSpaceToolkit/Physics/Environment.hpp

```diff
@@ -161,16 +190,34 @@ class Environment
     ///                     Contains Earth, Sun and Moon, with SPICE-based ephemeris.
     ///
     /// @code
-    ///                     Environment environment = Environment::Default() ;
+    ///                     Environment environment = Environment::Default();
     /// @endcode
+    /// @param              [in] setGlobalInstance True if the global environment instance should be set with the
+    /// default
     ///
     /// @return             Undefined environment
 
-    static Environment Default();
+    static Environment Default(const bool& setGlobalInstance = false);
+
+    /// @brief              Get the singleton instance of the environment
+    ///
+    /// @return             Shared pointer to the singleton instance
+
+    static Shared<Environment> AccessInstance();
+
+    /// @brief              Set the singleton instance of the environment
+    ///
+    /// @param              [in] instance Shared pointer to the new singleton instance
+
+    static void SetInstance(const Shared<Environment>& instance);
```

Comment:
Do we want to allow this to be changed externally once set?

I'd perhaps hold on that one for now (it's easy to add later)

Reply:
Hmmmmm I'm a bit confused. How would a user set it if we don't provide a setter?

Reply:
> How would a user set it if we don't provide a setter?

Maybe we can just add the `setToGlobal` to the actual arguments Environment Constructor too (defaulting to false). So you can create any Environment you want and set it to global on the spot:

```python
my_environment = Environment(my_objects, set_to_global=True)
```